test/src/postprocessors/InterfaceValueUOPPS.C

```diff
@@ -0,0 +1,51 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "InterfaceValueUOPPS.h"
+#include "InterfaceValueAverageUO.h"
+
+registerMooseObject("MooseTestApp", InterfaceValueUOPPS);
+
+template <>
+InputParameters
+validParams<InterfaceValueUOPPS>()
+{
+  InputParameters params = validParams<GeneralPostprocessor>();
+  params.addRequiredParam<UserObjectName>("user_object", "The name of the user object");
+  params.addClassDescription(
+      "Test Interface User Object postprocessor getting value from InterfaceValueAverageUO");
+  return params;
+}
+
+InterfaceValueUOPPS::InterfaceValueUOPPS(const InputParameters & parameters)
+  : GeneralPostprocessor(parameters),
+    _uo(getUserObject<InterfaceValueAverageUO>("user_object")),
+    _value_pp(0.)
+{
+}
+
+InterfaceValueUOPPS::~InterfaceValueUOPPS() {}
+
+void
+InterfaceValueUOPPS::initialize()
+{
+  _value_pp = 0;
+}
+
+void
+InterfaceValueUOPPS::execute()
+{
+  _value_pp = _uo.getValue();
```

Comment:
I am very concerned that this will not work correctly. I need to go over our user object and postprocessor execution, but here's what I think could happen: this gets executed after your user object gets executed. The value that you are retrieving from the user object will not have been finalized, e.g. we will not have done the necessary `gatherSums` across processors. So the final value that you have here in this postprocessor will not be equal to the final value in your user object.

A couple of questions:
1. I wanted to quickly verify my hypothesis that this won't quite be right by running one of your test files. I tried running `interface_value_user_object.i` but the solve did nothing; the initial non-linear residual was 0, so nothing ever really happened. Is that test really supposed to do nothing but report a bunch of zeros?
2. Do you really need the `InterfaceValueAverageUO` object or could you not just do all its work in your current `InterfaceValueUOPPS` object since it seems like the former is really just doing a computation that a postprocessor would do, e.g. it's building a single aggregate value?

Reply:
>  this gets executed after your user object gets executed. The value that you are retrieving from the user object will not have been finalized, e.g. we will not have done the necessary gatherSums

I noticed this behaviour and I don't understand why these has been conceived in this manner. 
This is also confusing when coding in general. Values from UO should always be retrieved after finalize and threadJoin (is there a way to do this?).

> the initial non-linear residual was 0, so nothing ever really happened. Is that test really supposed to do nothing but report a bunch of zeros?
I didn't realize it's not solving anything. wec ould try to see if it woprks properly simply leaving the same problem but changing IC of `u`,`v`, such that the interface kernel has to do something.
However It would be nice to have something more robust like a know solution at the interface for both variables.

> Do you really need the InterfaceValueAverageUO object or could you not just do all its work in your current InterfaceValueUOPPS object since it seems like the former is really just doing a computation that a postprocessor would do, e.g. it's building a single aggregate value?

I agree with you as long as updated finalize postprocessor values can be used inside other objects like Materials, Kernels ,etc. If this is the case we can get rid of `InterfaceValueAverageUO`. and rename `InterfaceValueUOPPS` -> `InterfaceValuePPS` and make all the calculations inside it. This would also imply removing some test.

Reply:
Yes postprocessor values can absolutely be used inside `Materials`, `Kernels`, etc. See `test/tests/src/kernels/DefaultPostprocessorDiffusion.C`. Let's rename `InterfaceValueUOPPS` to `InterfaceAverageValuePostprocessor` because we are computing a spatial average **and** we are leveraging your `InterfaceValue` capabilities. Sound good?

Reply:
@lindsayad  sounds great! This will also clean a bit the all PR and take away some confuision

Reply:
@lindsayad I complete removed the `InterfaceValueUOPPS` as the `InterfaceIntegralVariableAveragePostprocessor` was already doing this.
Now it should just be a matter of naming this postprocessor class